Source/Util/MapScanner.c

```diff
@@ -0,0 +1,159 @@
+// Copyright CircumScriptor and DarkNeutrino 2021
+#include "MapScanner.h"
+
+#include <Util/Alloc.h>
+#include <Util/Log.h>
+#include <Util/Utlist.h>
+#include <dirent.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+
+// Helper structure for sorting
+typedef struct map_entry {
+    char* name;
+    struct map_entry *next, *prev;
+} map_entry_t;
+
+static int
+is_directory(const char* path)
+{
+    struct stat st;
+    return (stat(path, &st) == 0 && S_ISDIR(st.st_mode));
+}
+
+static int
+file_exists(const char* path)
+{
+    struct stat st;
+    return (stat(path, &st) == 0 && S_ISREG(st.st_mode));
+}
+
+static int
+compare_map_names(map_entry_t* a, map_entry_t* b)
+{
+    return strcmp(a->name, b->name);
+}
+
+string_node_t*
+scan_maps_directory(const char* directory, uint8_t* map_count, int alphabetic)
```

Comment:
This is not how we do things. User NEEDS to be able to set the maps they want to use in the server. Not just scan the entire folder. Let the user define the maps they want and do the alphabetical and random on top of that.